tests/test_bookmarks.py

```diff
@@ -0,0 +1,145 @@
+import tap_tester.connections as connections
+import tap_tester.menagerie   as menagerie
+import tap_tester.runner      as runner
+import unittest
+import simplejson
+
+from base import TestSquareBase
+from test_client import TestClient
+
+class TestSquareIncrementalReplication(TestSquareBase):
+    START_DATE_FORMAT = "%Y-%m-%dT00:00:00Z"
+    START_DATE = "2020-06-01T00:00:00Z"
+
+    def get_properties(self, original = True):
+        return_value = {
+            'start_date' : '2020-06-01T00:00:00Z',
+            'sandbox' : 'true'
+        }
+
+        if original:
+            return return_value
+
+        return_value['start_date'] = self.START_DATE
+        return return_value
+
+
+    def name(self):
+        return "tap_tester_square_incremental_replication"
+
+    @staticmethod
+    def select_all_streams_and_fields(conn_id, catalogs):
+        """Select all streams and all fields within streams"""
+        for catalog in catalogs:
+            schema = menagerie.get_annotated_schema(conn_id, catalog['stream_id'])
+
+            connections.select_catalog_and_fields_via_metadata(conn_id, catalog, schema)
+
+    def run_sync(self, conn_id):
+        """
+        Run a sync job and make sure it exited properly.
+        Return a dictionary with keys of streams synced
+        and values of records synced for each stream
+        """
+        # Run a sync job using orchestrator
+        sync_job_name = runner.run_sync_mode(self, conn_id)
+
+        # Verify tap and target exit codes
+        exit_status = menagerie.get_exit_status(conn_id, sync_job_name)
+        menagerie.verify_sync_exit_status(self, exit_status, sync_job_name)
+
+        # Verify actual rows were synced
+        sync_record_count = runner.examine_target_output_file(
+            self, conn_id, self.expected_streams(), self.expected_primary_keys())
+        return sync_record_count
+
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    def test_run(self):
```

Comment:
Needs to look more like Trello with multiple syncs and updates/creates in between those syncs.